external/chromium/src/base/allocator/partition_allocator/partition_bucket.cc

```diff
@@ -167,6 +170,14 @@ uintptr_t ReserveMemoryFromGigaCage(pool_handle pool,
 #endif
 
   PA_DCHECK(!(reserved_address % kSuperPageSize));
+
+  size_t pkey = ::ia2_get_pkey();
```

Comment:
Is this what we want? Partition 0 should be shared, but what happens if the allocator needs to reserve more pages for this shared partition while running in a compartment? Don't we need to track the partition compartment as a field of that partition's metadata or something?

Reply:
You're right, this doesn't actually ever reserve memory for the shared partition. I don't think we need to store the partition's compartment in the metadata, just need make sure that the pkey used for allocations that trigger calls to this function gets propagated down to it.

Reply:
Actually here we have two options (1) use partition allocator for private and shared allocations, (2) use partition alloc for private allocations and glibc malloc for shared allocations. I originally intended to implement the former, but the latter is much easier to implement. I think the first option is ideal since it uses a single allocator, but I'm going to go with the latter for now to have something that we can use with nginx.